ui/app/upload_image/controller.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const multer = require('multer')
+const httpRequest = require('request')
+const returnError = require('../errors/index')
+const logger = require('../logging/index')
+const uploadFile = multer({
+    destination: './',
+    limits: {
+        fileSize: 5000000 // limit file size to 5MB
+    }
+})
+
+// handles HTTP POST for the route /upload-image
+const handleUploadImageRoute = [
+    uploadFile.single('file'),
+    (request, response, next) => {
+        const token = request.session.jwt
+        const file = request.file
+        const userId = request.session.userId
+        handleUploadingImage(file, userId, token).then((image) => {
```

Comment:
Let's call this param something other than `image` since it's really more like an `uploadResponse`